src/emulator/gui/src/game_selector.cpp

```diff
@@ -179,22 +190,105 @@ void draw_game_selector(GuiState &gui, HostState &host) {
+                                gui.game_selector.games.erase(games_index);
+                                if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
+                                    gui.current_background = gui.user_backgrounds[host.cfg.background_image];
+                                else if (!gui.user_backgrounds[host.cfg.background_image] && gui.current_background != 0)
+                                    gui.current_background = 0;
+                            } else
+                                LOG_ERROR("Deleting the game '{} [{}]' have failed.", game.title_id.c_str(), game.title.c_str());
+                            gui.delete_game_popup = false;
+                            ImGui::CloseCurrentPopup();
+                        }
+                        ImGui::SameLine();
+                        if (ImGui::Button("No", CONTEXT_BUTTON_SIZE)) {
+                            gui.delete_game_popup = false;
+                            ImGui::CloseCurrentPopup();
+                        }
+                        ImGui::EndPopup();
+                    } else
+                        ImGui::PopStyleColor();
                 }
+
             }
             ImGui::NextColumn();
-            ImGui::Selectable(game.title_id.c_str(), &selected[1], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, icon_size));
+            ImGui::Selectable(game.title_id.c_str(), &selected[1], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, static_cast<float>(icon_size)));
+            if (ImGui::IsItemHovered() && gui.title_id != game.title_id)
+                gui.title_id = game.title_id;
+            ImGui::OpenPopupOnItemClick("#game_context_menu", 1);
             ImGui::NextColumn();
-            ImGui::Selectable(game.app_ver.c_str(), &selected[2], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, icon_size));
+            ImGui::Selectable(game.app_ver.c_str(), &selected[2], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, static_cast<float>(icon_size)));
             ImGui::NextColumn();
             if (ImGui::IsItemHovered()) {
-                if (host.cfg.show_game_background) {
-                    if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
-                        gui.current_background = gui.user_backgrounds[host.cfg.background_image];
-                    else if (!gui.user_backgrounds[host.cfg.background_image] && gui.current_background != 0)
-                        gui.current_background = 0;
-                }
+                if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
```

Comment:
No checking if the value is the same here.

Reply:
Is old pr that i have set this for good reason, i d'ont change that

Reply:
I caught it now though. It's a number you're setting please it change it please please please

Reply:
? i no touch this, i have set this old check for no change value if no needed, is alrready good like that